lib/pure/options.nim

```diff
@@ -139,6 +139,34 @@ proc get*[T](self: Option[T], otherwise: T): T =
   else:
     otherwise
 
+template `or`*[T](x: Option[T]; y: T): T =
+  ## Alias for the ``get`` proc, but useful when chaining several `or`
+  ## operations. It might also be more readable to use this operator
+  ## than to use the get proc.
+  get(x,y)
+
+template `or`*[T](x, y: Option[T]): Option[T] =
+  let xx = x
```

Comment:
inline proc wouldn't need this for example

Reply:
Would an inline proc guarantee that y is not run if x has a value though? AFAIK it still runs procedures to generate an Option to pass in.

Reply:
@mratsim an linline proc evaluates all arguments before entering the function body. I explicity want to avoid that. The logical or operator also works like that. That is why you can write conditions like this:  `` if  a != nil and a.member > 123 ``. If ``or`` would evaluate all arguments first, this would crash when ``a`` is ``nil``.

Reply:
Right, that's a good reason :). I would add a comment `using templates to allow evaluation shortcuting` so that people don't change it to proc later then.

Reply:
@mratsim I have a testcase that covers that.